app/lib/shared/markdown.dart

```diff
@@ -8,8 +8,10 @@ import 'package:path/path.dart' as p;
 
 final Logger _logger = new Logger('pub.markdown');
 
-final List<m.BlockSyntax> _blockSyntaxes =
-    m.ExtensionSet.gitHub.blockSyntaxes.toList();
+final List<m.BlockSyntax> _blockSyntaxes = []
+  ..addAll(m.ExtensionSet.gitHub.blockSyntaxes)
+  ..add(const m.HeaderWithIdSyntax())
```

Comment:
The `markdown` lib 1.0 removed these from the github extension set, but I think github does generate the header ids and people rely on it for page-internal links, so I added those back here.